### PR TITLE
Extract Modified Behavior

### PR DESCRIPTION
**Pull request by terrellt**
This is composition over inheritance.

### CODE REVIEW

File: app/factories/term_factory.rb [hunk 1]
```diff
@@ -2,26 +2,37 @@ class TermFactory
...
+    end
+
+    def decorators
+      [
+        SetsModified,
+        TermWithChildren
+      ]
+    end
+
+    def repository
```
Comment (jechols) [inline app/factories/term_factory.rb]: +1

**Comment by jechols** [comment]
Looking at the factory, I think this is the kind of thing we need to make clear.  If you don't use a factory, what you're instantiating isn't going to run potentially important business logic.  I prefer to separate concerns nicely, so I'm not suggesting we fry factories - just that we should make it clear they're the entry point to terms.  Not sure how best to do that so new devs don't miss it.

File: app/factories/term_factory.rb (hunk 1 above)
Comment (terrellt) [inline app/factories/term_factory.rb]: Yeah, I was proud of that one

**Comment by terrellt** [comment]
> Looking at the factory, I think this is the kind of thing we need to make clear. If you don't use a factory, what you're instantiating isn't going to run potentially important business logic. I prefer to separate concerns nicely, so I'm not suggesting we fry factories - just that we should make it clear they're the entry point to terms. Not sure how best to do that so new devs don't miss it.

This is my concern too. It was especially important when behavior -changed- when I did this to Issued too. Suddenly if you do

```ruby
t = Term.new("test")
t.persist!

t = Term.find("test")
```
it raises ActiveTriples::NotFound, because there's nothing to persist (no automatically created properties), so it happily persisted nothing.

I don't know what the right way to deal with this is.

**Comment by jechols** [comment]
There is always the hacky way - make it so you can't instantiate a Term outside a factory, so the "default" approach very clearly won't work....  I think that would require some trickery that wouldn't be worth the pain, though.

**Comment by terrellt** [comment]
@jechols That defeats the purpose of decoration - a Term as a data model is totally valid.

**Comment by jechols** [comment]
True, but it's not the default as a new dev (or even a seasoned Rails dev not familiar with factories) might expect.

This might just be part of a bigger architecture documentation project.  It's certainly not a reason to hold up this PR.

**Pull request merged**